text/go/0001-go-bom.md

```diff
@@ -0,0 +1,163 @@
+# Go BOM Generation
+
+## Summary
+
+Add support for the generation of the Bill of Materials in the Go Language Family in order to provide entries for
+modules installed during the build process that can be found on the final app image.
+
+## Motivation
+
+As a part of an initiative to provide a Bill of Materials (BOM) for all the Paketo Language Family Buildpacks, in
+previous months
+the [Node.JS BOM Module](https://github.com/paketo-buildpacks/rfcs/blob/main/text/nodejs/0013-install-bom.md) was
+developed.
+
+We would like to add support for BOM generation in the Go language family. We would like to create a new buildpack to
+provide this. We do not anticipate the need for changes to other buildpacks in the language family.
+
+We propose to support generating BOMs only for projects which utilize vendoring via `go mod`. At this time we do not
+propose to support any other vendoring solutions
+(e.g. `dep` or files copied directly into the `vendor` directory without a
+`go.mod` file). Extending the new buildpack to support to other vendoring solutions could be a future RFC, if we
+discover a clear requirement for this.
+
+## Implementation
+
+### Tool details
+
+The Bill of Materials that will result should be an accurate accounting of all the modules that are defined and
+installed by `go modules`.
+
+There already exists a tool called [CycloneDX Gomod](https://github.com/CycloneDX/cyclonedx-gomod) which is an official
+CycloneDX supported-tool that creates valid CycloneDX BOMs for Go applications. CycloneDX is a BOM format supported by
+the OWASP Foundation, and is a serious contender (at the time of this RFC creation) as a format we will eventually
+support.
```

Comment:
```suggestion
the OWASP Foundation, and is a format that we plan to support per the recently merged [upstream CNB RFC](https://github.com/buildpacks/rfcs/blob/main/text/0095-sbom.md).
```

Reply:
Side note, now that the upstream CNB RFC has been merged in, do we need to add onto this initiative to include transitioning the BOM format from the legacy CNB format to SPDX/CycloneDX in a top-level RFC addendum? Or is this something we will tackle later? cc @ForestEckhardt @simonjjones

Reply:
The RFC was approved, but there is no current support in the lifecycle for the new SBOM format. We can move forward with an RFC, but implementations would need to wait until there was lifecycle support.